packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -556,7 +556,9 @@ public ResponseEntity<Void> createDocumentFromXDD(@RequestBody final AddDocument
 			DocumentAsset documentAsset = createDocumentAssetFromXDDDocument(
 					document, userId, extractionResponse.getSuccess().getData(), summaries, permission);
 			if (filename != null) {
-				documentAsset.getFileNames().add(filename);
+				if (!documentAsset.getFileNames().contains(filename)) {
```

Comment:
Making file names have to be unique is a new limitation. Would we want to throw back some sort of responseStatusException?

Reply:
This is to prevent duplicates since sometimes the frontend sets it and sometimes the backend sets it.